core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -215,6 +215,14 @@ class HadoopMapReduceCommitProtocol(
           }
           fs.rename(new Path(stagingDir, part), finalPartPath)
         }
+        // move the _SUCCESS file to the final location
+        fs.globStatus(new Path(s"${stagingDir.toString}/*")).foreach {
```

Comment:
This is woefully inefficient as it's listing the entire directory tree to look for one file whose name is known. Better
```scala
          val success = new Path(stagingDir, "_SUCCESS")
          if (fs.isFile(success)) {
            fs.rename(success, new Path(path, "_SUCCESS"))
          }
```